packages/utils-create-server/src/index.ts

```diff
@@ -3,19 +3,25 @@ import { spawn, ChildProcess, SpawnOptions } from 'child_process';
 import map = require('lodash/map');
 import reduce = require('lodash/reduce');
 
-import { Message, ServerConfiguration } from './types';
+import * as isCI from 'is-ci';
+
+import { IServer, Message, ServerConfiguration } from './types';
 import { replacer, reviver } from './buffer-serialization';
 
+import { Server as ServerProcess } from './server';
+
 export { ServerConfiguration };
+export { IServer };
 
 // Do all magic wit process here and maybe have a wrapper
 
 type ServerOptions = {
     configuration?: ServerConfiguration;
     isHTTPS?: boolean;
+    sameThread?: boolean;
 }
 
-export class Server {
+class IndependentServer implements IServer {
```

Comment:
What's the difference between `IndependentServer` and `Server`? Can you add some comments?

Reply:
`Server` is a static class with a factory that get exported so everyone has to go through here.